### ISSUE TITLE
Stormcraft corporation

### ISSUE THREAD

**Issue by vincentneko** [bafolts/terraforming-mars#205]
I have 2 questions regarding this corporation.

1/ Should it be added to the Venus Next corporation pool ?

2/ The Effect of this card is : Floaters on this card may be used as 2 heat each.
Do you have an idea on how to implement this without specific code each time heat can be used ?

I see the following cases for heat usage :
- Raise Temp with 8 heat feature
- CARETAKER CONTRACT card
- LOCAL HEAT TRAPPING card
-

**Comment by bafolts** [comment]
Adding `Stormcraft` to a `venusNext/colonies` namespace would make sense to me as that card can only be used with both venus and the colonies expansions.

The expansion cards are more specific than the original cards. My major concern with pulling so many dependencies into `Game` is that the dependency list will get very large when from the `Game` we only really care about the name of the card. Perhaps we could create an enum of all card names. This enum could be used in `src/HTML_data` as well as in each card. This way from `Game` or `Player` we would only need a dependency to the enum and not the instance of the cards. This would at least make the check for if a player has the card easier and we could add methods such as:

```
public isCorporation(corporationName: CorporationName): boolean
public cardIsInEffect(cardName: CardName): boolean
```

This should make unit testing of `Game` or `Player` still easy as a unit test only need to pass in the corporation name and not an instance of the card.

Besides the semantics of where the code goes the user is going to need to select two amounts. They will need to select how many of their floaters from their card they want to use and how much of their normal heat they want to use. I would use an `AndOptions` with two `SelectAmount` components for that part. The modification I plan to make to upgrade the `AndOptions` to only show one save button should help for this usage as well.

In summary. Introducing an enum for the corporation card and card names would be good enough for me. Then we can replace all the instances where we create an instance of the card to pull it's name with the enum value. We can introduce the card name enum values as we use them for the time being and make an issue to backfill them.

**Comment by vincentneko** [comment]
Thanks for the update.
I tried to put this in perspective, I've commited a change into cardCostRework branch for Pets and Protected Habitats, please have a look and let me know if it is what you have in mind.

**Comment by bafolts** [comment]
Yes. That change looks good. I would capitalize the enum as CardList and
use a name like CardName instead of CardList as this object will not be a
list of cards but a card name.

**Comment by vincentneko** [comment]
Incoming PR, please review.

**Comment by vincentneko** [comment]
Implemented
